### PR TITLE
feat(dividends): group dividends by issuer in the report (closes #200)

### LINKED ISSUE #200: ✨ Group dividends by issuer in the report (match Spanish brokers' fiscal certificates)
## Context

DeclaRenta currently lists dividends **one row per cash transaction**
(one entry per quarterly/monthly payment, one entry per withholding
adjustment). Spanish brokers with a convenio with the AEAT
(MyInvestor, ING, Renta 4, etc.) hand the user a year-end fiscal
certificate **aggregated per issuer**: one block per company / ISIN
showing the annual gross dividend total, total withholding, and (often)
a sub-table with the individual payments inside.

That layout is also what the AEAT Renta Web "Alta Capital mobiliario"
form is designed for — one record per company with the annual total.
Right now a user with twenty quarterly dividends across five companies
has to mentally aggregate them to type the per-company totals into
Renta Web, or paste twenty rows where five would do.

Tax math is unchanged: Casilla 0029 is a single total. This is purely
a presentation / user-comfort improvement, in line with the
"Correctness with User Comfort" principle in `CLAUDE.md`.

## Proposal

Show dividends grouped per issuer (one block per company) in the
Modelo 100 → Casilla 0029 detail card, the CSV export, and the PDF.
Each issuer block should carry:

- Annual gross dividend in EUR
- Annual withholding in EUR
- Withholding country
- Number of payments
- (Optional) collapsible per-payment detail underneath

Keep the existing per-payment data accessible — collapsing is for the
default view, not a destructive change.

### Today vs proposed

Concrete shape of the gain for a user holding AAPL (4 quarterly
payments) and KO (4 quarterly payments) in 2025:

**Today** — 8 rows, user mentally aggregates per company before
typing the totals into Renta Web:

| ISIN | Symbol | Date       | Gross (EUR) | Withholding (EUR) |
|------|--------|------------|-------------|-------------------|
| US…  | AAPL   | 2025-02-13 | 24.50       | 3.68              |
| US…  | AAPL   | 2025-05-15 | 24.50       | 3.68              |
| US…  | AAPL   | 2025-08-14 | 25.00       | 3.75              |
| US…  | AAPL   | 2025-11-13 | 25.00       | 3.75              |
| US…  | KO     | 2025-04-01 | 19.40       | 2.91              |
| US…  | KO     | 2025-07-01 | 19.40       | 2.91              |
| US…  | KO     | 2025-10-01 | 19.40       | 2.91              |
| US…  | KO     | 2025-12-15 | 19.40       | 2.91              |

**Proposed** — 2 rows by default, drill-down preserves the detail:

| ISIN | Symbol | Country | Payments | Gross total | Withholding total |
|------|--------|---------|----------|-------------|-------------------|
| US…  | AAPL   | US      | 4        | 99.00       | 14.86             |
| US…  | KO     | US      | 4        | 77.60       | 11.64             |

## Out of scope

- The total amount declared in Casilla 0029 doesn't change.
- The double-taxation deduction (Casilla 0588) doesn't change.
- Modelo 720 / 721 are unaffected.

## References

- AEAT Renta Web — "Alta Capital mobiliario" supports one record per
  company with the annual total.
- DeGiro / MyInvestor annual Fiscal Notes group distributions per
  asset.
- Modelo 193: the broker-side declaration that the per-issuer
  certificates are derived from.
- `CLAUDE.md` → "Correctness with User Comfort" + "Actionable
  Explanations".

### PR DESCRIPTION
**Pull request by GeiserX**
## Summary

Closes #200. Spanish brokers' year-end fiscal certificates and the AEAT Renta Web **"Alta Capital mobiliario"** form are organized **one record per company** with the annual total. DeclaRenta listed dividends one row per payment, forcing a user with, say, 4 AAPL + 4 KO quarterly dividends to mentally aggregate 8 rows into the 2 per-company totals Renta Web wants.

This adds a per-issuer view across the Casilla 0029 detail card, CSV, and PDF, **preserving** the per-payment detail as a drill-down.

### Today → proposed
8 payment rows → 2 issuer rows (AAPL: 4 payments, €99.00 gross / €14.86 WHT; KO: 4 payments, €77.60 / €11.64), each expandable to its individual payments.

## Design

`groupDividendsByIssuer(entries)` in `generators/casillas.ts` — a **pure render-layer helper** mirroring `computeCasillaBlocksWithFx`, never stored on `TaxSummary`.

- **Key** = `(isin || symbol || description) + withholdingCountry`. Country is part of the key so each block maps **1:1 to a Casilla 0588 country line** (a rare same-ISIN/two-country case shows as two honest, reconcilable blocks rather than one block silently mixing countries). Empty-ISIN issuers fall back to symbol/description so distinct coins never merge. Mixed currencies within a group → `currency: "—"`.
- Sums gross/withholding EUR, carries `paymentCount` + the per-payment `payments[]` for drill-down. Sorted gross DESC, key ASC (deterministic).

Wired into:
- **Web** Casilla 0029 card: per-issuer rows + collapsible `<details>` per-payment drill-down.
- **CSV**: a new `# DIVIDENDOS POR EMISOR` section; the per-payment `# DIVIDENDOS` section is **kept unchanged** so neither total is double-summed on a naive `SUM`.
- **PDF** (CLI) + **web PDF**: per-issuer summary.
- **i18n**: `table.payments`, `casilla.dividends_by_issuer`, `casilla.dividends_per_payment` in all 5 locales.

## Tax-math safety (presentation-only)

`grossIncome` (Casilla 0029) and `calculateDoubleTaxation` (Casilla 0588) are computed from the raw per-payment `entries` **independently** — the grouping helper is never consumed by tax math. The load-bearing invariant `Σ group.grossTotalEur === grossIncome` is asserted in tests; 0588 and Modelo 720/721 are untouched.

This was scoped via a research panel (codebase + architecture + AEAT primary-source verification + adversarial critique). The (issuer+country) key and separate-CSV-section choices directly address the two critical risks the critic raised (empty-ISIN merge, ISIN-only country collision).

## Test plan
- [x] `npm run typecheck`, `npm run lint`, `npm test` — **1278 tests pass** (+8: total-invariance, country split, empty-ISIN fallback, mixed-currency, CSV per-issuer section).
- [x] `npm run build`
- [ ] Browser smoke-test of the Casilla 0029 card drill-down — not run in this environment; reuses the existing `<details>`/`.detail-table` styled patterns and is unit-tested.

## Summary by CodeRabbit

* **New Features**
  * Dividends are grouped by issuer + withholding country across UI, PDF and CSV exports, showing per-issuer payment count and aggregated gross/withholding totals with expandable drill-down to individual payments.

* **Documentation**
  * Added guidance describing the presentation-only dividend grouping and its reconciliation constraints with tax totals.

* **Tests**
  * Added tests verifying grouping behavior, reconciliation with tax figures, sorting, currency-mixing, and edge cases.

### CODE REVIEW

File: src/generators/casillas.ts [hunk 1]
```diff
@@ -156,3 +156,62 @@ export function computeCasillaBlocksWithFx(report: TaxSummary): CasillaBlocks {
...
+    // Distinct currencies within a group are still summed correctly (all EUR),
+    // but no single currency label is meaningful → mark as mixed.
+    if (g.currency !== "—" && g.currency !== d.currency) g.currency = "—";
+  }
+
+  for (const g of groups.values()) {
+    g.payments.sort((a, b) => a.payDate.localeCompare(b.payDate));
+  }
+
+  // Largest issuer first (how a user scans a fiscal certificate); key asc breaks
```
Comment (lucasfarre) [inline src/generators/casillas.ts]: Have you considered ordering it alphabetically by issuer instead?

**Review by lucasfarre — APPROVED** [review]
Thanks @GeiserX for the quick feature! I tried the branch and works as expected :)

**Comment by GeiserX** [comment]
@coderabbitai review
